tests/e2e/controlplane/control_plane_test.go

```diff
@@ -232,6 +238,37 @@ spec:
 							ShouldNot(ContainSubstring("Reconciliation done"), "Istio Operator is continuously reconciling")
 						Success("Istio Operator stopped reconciling")
 					})
+
+					It("can be deployed bookinfo with sidecar inyection", func(ctx SpecContext) {
+						By("Applying bookinfo YAML")
+						Expect(deployBookinfo(version)).To(Succeed(), "Error deploying bookinfo")
+						Success("Bookinfo deployed")
+
+						By("Waiting for bookinfo pods to be ready")
+						bookinfoPods := &corev1.PodList{}
+						cl.List(ctx, bookinfoPods, client.InNamespace(bookinfoNamespace))
+						Expect(bookinfoPods.Items).ToNot(BeEmpty(), "No pods found in bookinfo namespace")
+
+						for _, pod := range bookinfoPods.Items {
+							Eventually(common.GetObject).WithArguments(ctx, cl, kube.Key(pod.Name, bookinfoNamespace), &corev1.Pod{}).
+								Should(HaveCondition(corev1.PodReady, metav1.ConditionTrue), "Pod is not ready")
+						}
+						Success("Bookinfo pods are ready")
+
+						By("Checking the sidecar injection version match the expected version")
+						for _, pod := range bookinfoPods.Items {
+							sidecarVersion, err := getProxyVersion(pod.Name, bookinfoNamespace)
+							Expect(err).To(Succeed(), "Error getting sidecar version")
+							Expect(sidecarVersion).To(ContainSubstring(version.Version), "Sidecar injection version does not match the expected version")
```

Comment:
Is checking for substring enough here?

Reply:
There is an issue here, because for example for latest the version that we will get from the sidecar will be something like `1.24-alpha-xxxxxxx`, because of that I'm checking the substring. For released version will work with an `equal` but for latest will not work